File: code-coaches/medium/convert-us-date-to-eu-date/ConvertUsDateToEuDate.java
import java.util.Scanner;
import java.util.HashMap;

public class ConvertUsDateToEuDate {
	
	public static void main(String[] args) {
		HashMap<String, String> monthToNum = new HashMap<String, String>() {
			{
				put("January", "1");
				put("February", "2");
				put("March", "3");
				put("April", "4");
				put("May", "5");
				put("June", "6");
				put("July", "7");
				put("August", "8");
				put("September", "9");
				put("October", "10");
				put("November", "11");
				put("December", "12");
			}
		};
		
		Scanner input = new Scanner(System.in);
		String usDate = input.nextLine();
		String euDate = "";

		if (usDate.contains("/")) {
			String[] usDateSplit = usDate.split("/");
			euDate = usDateSplit[1] + "/" + usDateSplit[0] + "/" + usDateSplit[2];
		} else {
			String[] usDateSplit = usDate.split(" ");
			String month = monthToNum.get(usDateSplit[0]);
			String day = usDateSplit[1].substring(0, usDateSplit[1].length() - 1); // remove ','
			euDate = day + "/" + month + "/" + usDateSplit[2];
		}
		
		System.out.println(euDate);

		input.close();
	}

}

File: code-coaches/medium/convert-us-date-to-eu-date/convert_us_date_to_eu_date.cpp
#include <iostream>
#include <map>
#include <vector>
#include <sstream>

void split(std::string const &str, const char delim, std::vector<std::string> &out)
{
	std::stringstream ss(str);
	std::string segment;

	while (getline(ss, segment, delim))
	{
		out.push_back(segment);
	}
}

std::map<std::string, std::string> month_to_num = {
	{"January", "1"},
	{"February", "2"},
	{"March", "3"},
	{"April", "4"},
	{"May", "5"},
	{"June", "6"},
	{"July", "7"},
	{"August", "8"},
	{"September", "9"},
	{"October", "10"},
	{"November", "11"},
	{"December", "12"},
};

std::string get_eu_date(std::string us_date)
{
    std::string eu_date;
    std::vector<std::string> us_date_split;

	if (us_date.find("/") != std::string::npos)
	{
		split(us_date, '/', us_date_split);
		eu_date = us_date_split[1] + "/" + us_date_split[0] + "/" + us_date_split[2];
	}
	else
	{
		split(us_date, ' ', us_date_split);

		std::string month = month_to_num.at(us_date_split[0]);

		us_date_split[1].pop_back();
		std::string day = us_date_split[1];

		eu_date = day + "/" + month + "/" + us_date_split[2];
	}

	return eu_date;
}

int main()
{
	std::string us_date;
	getline(std::cin, us_date);

	std::cout << get_eu_date(us_date);

	return 0;
}
